### ISSUE TITLE
Question about reproduction

### ISSUE THREAD

**Issue by JonnyS1226** [ioanacroi/qb-norm#2]
Thanks for your great work!
And I have two questions about reproducing results.
1. I have a question about 'Adversarial Domain', that is, how to generate it from MSRVTT? I tried to select the samples which are among top 5k minimum retrieved values of each query, but cannot reproduce the results of the paper, on the setting of MSRVTT full and TT-CE.
2. I have a question about querybanks for clip2video. That is, i generate querybanks directly from CLIP2Video inference , and finally get the train_test shape as [180000, 1000]. But after qbnorm, the result is {'R1': 46.4, 'R5': 72.6, 'R10': 82.5, 'R50': 94.4, 'MedR': 2.0, 'MeanR': 14.1, 'geometric_mean_R1-R5-R10': 65.3}
I think i may have some misunderstandings? Really appreciate it if you can reply.

**Comment by ioanacroi** [comment]
Hi,

1. So, we constructed the Adversarial querybank for MSRVTT by taking 5000 queries that retrieve the least videos. So, we computed the most retrieved video and put the queries from that video in the querybank, then we computed the second most retrieved video and put those queries in the querybank and so on until we reach 5000 queries.
2. What temperature (beta) did you use? We will try to upload the similarity matrices for CLIP2Video also.

Thanks,
Ioana

**Comment by JonnyS1226** [comment]
Thanks for your help! And I have reproduced results according to your first advice.
In clip2video, beta is set to 0.5. Looking forward to the matrices!

**Comment by HanielF** [comment]
@JonnyS1226 Have you reproduced the result in clip2video which is reported as 47.2 on the R@1? Can you share your reproduction process？

**Comment by ioanacroi** [comment]
Hi,

We published here the sims matrices for CLIP2Video (https://www.robots.ox.ac.uk/~vgg/research/teachtext/QB-Norm/msrvtt-1ka-clip2video-sims.tar.gz). For the final experiments we used beta = 1/1.99, as stated in the paper.

Cheers,
Ioana

**Comment by JonnyS1226** [comment]
> Hi,
> 
> We published here the sims matrices for CLIP2Video (https://www.robots.ox.ac.uk/~vgg/research/teachtext/QB-Norm/msrvtt-1ka-clip2video-sims.tar.gz). For the final experiments we used beta = 1/1.99, as stated in the paper.
> 
> Cheers, Ioana

Thanks!

**Comment by JonnyS1226** [comment]
> @JonnyS1226 Have you reproduced the result in clip2video which is reported as 47.2 on the R@1? Can you share your reproduction process？

I just use clip2video inference (text in training set and video in test set), get the final train_test_sim, and the results is sensitive to beta. The author has published the sims of clip2video~

**Comment by HanielF** [comment]
> I just use clip2video inference (text in training set and video in test set), get the final train_test_sim, and the results is sensitive to beta. The author has published the sims of clip2video~

Thanks for your reply. I have tested with the train_test sim matrix of clip2video. And as you pointed out, this result is so sensitive to the beta that I mistakenly thought I made a mistake somewhere.

**Comment by HanielF** [comment]
> Hi,
> 
> We published here the sims matrices for CLIP2Video (https://www.robots.ox.ac.uk/~vgg/research/teachtext/QB-Norm/msrvtt-1ka-clip2video-sims.tar.gz). For the final experiments we used beta = 1/1.99, as stated in the paper.
> 
> Cheers, Ioana

@ioanacroi  Thanks for uploading the sim matrix of clip2video. I have two questions about it.

1. Why the shape of train_test sim matrix of clip2video is `(130260, 1000)`, instead of `(180000, 1000)`? 
2. I notice that beta is `1/1.99` instead of `0.5` mentioned in paper. How to find out the best beta for my own model? Do some brute force search?

**Comment by ioanacroi** [comment]
Hi,

Thanks you both for the questions!
1. We used the original training MSR-VTT split when extracting the sims matrix for CLIP2Video (due to some time constraint at that point). [JonnyS1226](https://github.com/JonnyS1226) Did you find that using the whole (180000, 1000) matrix produce lower numbers?
2. We updated the paper on 6 April and it reflects the 1/1.99 value. Since the QB-Norm algorithm runs fast, we afford to do some brute force. For your own dataset I suggest to do a general search first to find a good value and then to do a fine-grain search around that value. In our paper we strived for consistency and tried to use a global beta for all datasets. So, if you are interested, for most datasets you can find a better temperature that yields better results.

Thanks,
Ioana

**Comment by JonnyS1226** [comment]
> 

yes, when use the (180000, 1000) train_test, my best reproduced results R@1 is 46.9, where beta is 0.55, but this is already a big improvement.

**Comment by ioanacroi** [comment]
Hi,

Thanks for pointing that out! We have not tested with the full 9k training queries. 
The reason why beta is around 0.5 for CLIP models(clip2video and clip4clip) is because the final similarity matrix is scaled by the methods which is differently than CE.

Ioana

**Comment by HanielF** [comment]
@ioanacroi Thanks for your patient explanation. 

I didn't test the `(180000, 1000)` sim matrix on clip2video but for my own model, which did not achieve some improvement. It confused me for days after I tested different ways to construct the query bank, such as sampling 5000 from training set and using the whole 180000 text queries in training set. 

I also tried a brute force search for beta, but its best results also didn't beat the original model's results, which disappointed me a lot.
